### ISSUE TITLE
ldap group mappings

### ISSUE THREAD

**Issue by schen59** [grafana/grafana#3471]
Right now ldap group mappings is available by hard coding the mapping to the ldap config file. This required us to know the ldap information/org information about the user before hand. In multi-tenancy situation a ldap user can belong to multiple orgs and we can create new org on the fly and add new members to the org. In the case, the static group mapping in the config file may not sufficient enough. Can we add some support like expose some rest api to allow the admin to create new ldap group mapping dynamically?

**Comment by torkelo** [comment]
Ldap is not really designed for multi tenancy scenarios.

**Comment by schen59** [comment]
Is it possible to extend Ldap to support multi tenancy in the future? Coz grafana support multi tenancy I think it makes sense for Ldap to support it as well.

**Comment by torkelo** [comment]
maybe, but the multi tenancy support is mostly built for large scale saas installs where LDAP is not really used much

**Comment by madshall** [comment]
I would disagree, lots of companies which use LDAP might want to have a single instance of Grafana serving multiple teams with their own orgs (workspaces).

**Comment by torkelo** [comment]
Well a Grafana organization was not built to handle different teams/workspaces but multi tenant setups, meaning completly different companies. 

But not saying its wrong to use organization that way because there us currently no other way

**Comment by zstyblik** [comment]
Just to let you know that "our" use-case is exactly the same as @madshall's.

**Comment by haidaraM** [comment]
Same here. The LDAP config currently is not really flexible. For instance, impossible to map a LDAP group to multiple organizations in Grafana.

**Comment by bendem** [comment]
Mostly the same use case here.

We have developers working on projects for different clients and we've been using an organisation per client. We'd like our developers to have access to all organisations and our clients to only have access to their own org.

Atm, all the developers share the admin login because there is no way to assign them to multiple orgs. That would be a great addition.

**Comment by bendem** [comment]
Oh, we are playing the game of "ping the ticket every x days so it's not closed" here as well.

**Comment by exolicious** [comment]
i implemented this exact feature for the generic oauth connector. Maybe this could be upvoted by others so the maintainers finally have a look at it.

https://github.com/grafana/grafana/pull/93402#issue-2531203178

I could maybe also look to implement this feature for ldap aswell, if enough people comment on it.

**Comment by azlux** [comment]
Hi @exolicious , I think you have produce the right PR for that. Waiting for the grafana teams.
